src/algebra/monoid_algebra.lean

```diff
@@ -717,6 +717,42 @@ by rw [of, monoid_hom.coe_mk, ← finsupp.supported_eq_span_single, finsupp.mem_
 
 end span
 
+section opposite
+
+open finsupp opposite
+
+variables [semiring k]
+
+-- simps doesn't do the right thing for `symm_apply`
+/-- The opposite of an `add_monoid_algebra R I` is additively equivalent to
+the `add_monoid_algebra Rᵒᵖ I` over the opposite ring, taking elements to their opposite. -/
+@[simps apply] protected noncomputable def op_add_equiv :
+  (monoid_algebra k G)ᵒᵖ ≃+ monoid_algebra kᵒᵖ G :=
+opposite.op_add_equiv.symm.trans (finsupp.map_range.add_equiv opposite.op_add_equiv)
+
+lemma op_add_equiv_single (i : G) (r : k):
+  monoid_algebra.op_add_equiv (op (single i r)) = single i (op r) :=
+by simp
```

Comment:
Do you have an application in mind for this definition? If not, I'd be inclined to remove it - it's straightforward to construct from the pieces, and it's not clear that `G` is more desirable than `Gᵒᵖ` on the RHS.